components/content/CampaignModal.tsx

```diff
@@ -0,0 +1,74 @@
+import {AntDesign} from '@expo/vector-icons';
+import {Button} from 'components/content/Button';
+import {Center, View, VStack} from 'components/core';
+import {BodyBlack} from 'components/text';
+import {useToggle} from 'hooks/useToggle';
+import React, {useCallback} from 'react';
+import {Image, Modal, TouchableOpacity, TouchableWithoutFeedback} from 'react-native';
+import {SafeAreaView} from 'react-native-safe-area-context';
+import {colorLookup} from 'theme';
+
+export interface CampaignModalProps {
+  onAction?: () => void;
+}
+
+export const CampaignModal: React.FC<CampaignModalProps> = ({onAction = () => undefined}) => {
+  const [showModal, {off: hideModal}] = useToggle(true);
+  const activateCampaign = useCallback(() => {
+    hideModal();
+    onAction();
+  }, [hideModal, onAction]);
+
+  return (
+    // Pressing the Android back button dismisses the modal
```

Comment:
We had some feedback that the swipe-to-go-back gesture was not working to exit our observation filter modals - does the approach you're using here fix that as well? If so - let's capture that fix in an issue to update the other modals as well!

Reply:
#640